### PR TITLE
DO-128: Fixed the issue with showing "No Data Available" while loading.

### PR DESCRIPTION
**Pull request by dhruvlahoti**

### CODE REVIEW

File: src/services/member/reducer/member.js [hunk 1]
```diff
@@ -107,13 +114,16 @@ const getUpdatePasswordMessage = state => state.member.updatePassword.message;
 
 const getAppliedFilter = state => state.member.appliedFilter;
 
+const getIsMemberListExecuted = state => state.member.isMemberListExecuted;
```
Comment (mohitsaladiya) [inline src/services/member/reducer/member.js:117]: Please remove this.

File: src/components/routes/Member/AuthenticatedMember/MemberListViewRoute/MemberListViewRoute.jsx [hunk 2]
```diff
@@ -24,6 +24,7 @@ const {
   getMemberDetailsByIdAction,
 } = MemberModule.actions;
 const {
+  getIsMemberListExecuted,
```
Comment (mohitsaladiya) [inline src/components/routes/Member/AuthenticatedMember/MemberListViewRoute/MemberListViewRoute.jsx:27]: No need of this.

File: src/components/routes/Member/AuthenticatedMember/MemberListViewRoute/MemberListViewRoute.jsx [hunk 3]
```diff
@@ -157,12 +158,15 @@ class MemberListViewRoute extends Component {
         />
       );
     }
-    // TODO create an common alert box for all screen/route.
-    return (
-      <div className="mbr-list-no-data">
-        <Paragraph text="No Data Available" />
-      </div>
-    );
+    if (this.props.isMemberListExecuted) {
```
Comment (mohitsaladiya) [inline src/components/routes/Member/AuthenticatedMember/MemberListViewRoute/MemberListViewRoute.jsx:161]: usePromiseTracker(); method. For reference See Loader.jsx. 
However I feel this is not the accurate way to do this. But we can use this until we find an absolute solution.

File: src/services/member/actions/member.js [hunk 4]
```diff
@@ -200,3 +200,11 @@ export const setAppliedFilter = appliedFilter => ({
   type: 'SET_APPLIED_FILTER',
   appliedFilter,
 });
+
+/**
+ *
+ * @return {{response: *, type: string}}
+ */
+export const executeMemberListAction = () => ({
```
Comment (mohitsaladiya) [inline src/services/member/actions/member.js:208]: No need of this.

File: src/services/member/reducer/member.js [hunk 5]
```diff
@@ -9,14 +9,21 @@ const defaultState = {
 //  Ideally we should fetch this data in component only via async/await calls.
 const member = (state = defaultState, action) => {
   switch (action.type) {
+    case 'EXECUTING_MEMBER_LIST':
+      return {
+        ...state,
+        isMemberListExecuted: false,
```
Comment (mohitsaladiya) [inline src/services/member/reducer/member.js:15]: We can remove this.

File: src/services/member/sagas/member.js [hunk 6]
```diff
@@ -30,6 +31,7 @@ const { getAPIConfig } = apiSelectors;
 export function* getMembersListSaga(action) {
   try {
     const { params } = action;
+    yield put(executeMemberListAction());
```
Comment (mohitsaladiya) [inline src/services/member/sagas/member.js:34]: Remove this.

File: src/services/member/reducer/member.js (hunk 1 above)
Comment (dhruvlahoti) [reply src/services/member/reducer/member.js]: Done

File: src/components/routes/Member/AuthenticatedMember/MemberListViewRoute/MemberListViewRoute.jsx (hunk 2 above)
Comment (dhruvlahoti) [reply src/components/routes/Member/AuthenticatedMember/MemberListViewRoute/MemberListViewRoute.jsx]: Done

File: src/components/routes/Member/AuthenticatedMember/MemberListViewRoute/MemberListViewRoute.jsx (hunk 3 above)
Comment (dhruvlahoti) [reply src/components/routes/Member/AuthenticatedMember/MemberListViewRoute/MemberListViewRoute.jsx]: Can't use as it is a hook and can be used only in functional component

File: src/services/member/actions/member.js (hunk 4 above)
Comment (dhruvlahoti) [reply src/services/member/actions/member.js]: Done

File: src/services/member/sagas/member.js (hunk 6 above)
Comment (dhruvlahoti) [reply src/services/member/sagas/member.js]: Done

**Pull request merged**